test/e2e_node/hugepages_test.go

```diff
@@ -60,22 +48,29 @@ func makePodToVerifyHugePages(baseName string, hugePagesLimit resource.Quantity)
 			RestartPolicy: apiv1.RestartPolicyNever,
 			Containers: []apiv1.Container{
 				{
-					Image:   busyboxImage,
+					Image:   imageutils.GetE2EImage(imageutils.HugePageTester),
 					Name:    "container" + string(uuid.NewUUID()),
 					Command: []string{"sh", "-c", command},
+					Resources: apiv1.ResourceRequirements{
+						Limits: apiv1.ResourceList{
+							apiv1.ResourceName("cpu"):                                resource.MustParse("10m"),
+							apiv1.ResourceName("memory"):                             resource.MustParse("100Mi"),
+							apiv1.ResourceName("hugepages-" + hugePageSize.String()): totalHugePageMemory,
+						},
+					},
 					VolumeMounts: []apiv1.VolumeMount{
 						{
-							Name:      "sysfscgroup",
-							MountPath: "/tmp",
+							Name:      "hugetlb",
+							MountPath: "/hugetlb",
 						},
 					},
 				},
 			},
 			Volumes: []apiv1.Volume{
 				{
-					Name: "sysfscgroup",
+					Name: "hugetlb",
```

Comment:
we should still have a test container that mounts the sys/fs/cgroup volume into the pod so we can inspect the actual cgroup settings are applied since the intention is to validate that usage of pre-allocated hugepages have the limit enforced.

Reply:
if i recall, access to hugepages when consumed by a jvm is not through a volume, so having a test that still read the proper values were written that we expected into the cgroup tree is useful and consistent with our testing for all other cgroup controllers (cpu, memory, pids).

Reply:
Yee, did think about verifying it like the previous test did, so I guess I can keep the old one as well. 

> if i recall, access to hugepages when consumed by a jvm is not through a volume, so having a test that still read the proper values were written that we expected into the cgroup tree is useful and consistent with our testing for all other cgroup controllers (cpu, memory, pids).

Yee. The testing progam tests both via the mounted volum, and via normal mmap with a specified page size(see:
https://github.com/kubernetes/kubernetes/pull/78495/files#diff-e2d845a4981d87782f433ee65956641cR51). (I assume that is the way the jvm does it)

> tree is useful and consistent with our testing for all other cgroup controllers (cpu, memory, pids).

Didn't think about that, keeping the old test then :smile:

Reply:
Readded the old test which was checking the cgroup file now, @derekwaynecarr.